### DISCUSSION TITLE
Is git guardian a scam?

### DISCUSSION THREAD

**Discussion by vinyashegde** [community/community#32833] [Actions] [answered]
Got a mail from git gaurdian it said my generic key is exposed and there's a fix the leak button

**Comment by DenverCoder1** [comment, marked as answer]
[GitGuardian](https://www.gitguardian.com/) seems to be a legitimate third-party service that scans code on GitHub and notifies of vulnerabilities.

As always, you should double check the links in the email before clicking any. It's possible they found a false positive, but in any case, you should check if there is in fact a key exposed in your code and regenerate it if it is the case.

API keys should not be posted on GitHub, you should instead create local environment files for your secrets and write code that reads the secrets out of your configuration files. If it is a public project, it is recommend to instruct users how to go about creating their own configuration file to run the project and include an example with the keys removed. A `.gitignore` can be used to make sure the configuration files do not get tracked on GitHub.

**Reply by vinyashegde** [reply]
thanks for answering my doubt

**Reply by lokesh-esds** [reply]
@DenverCoder1 

I have also received a mail like this attached below for reference , 
For the repo for which the mail has been received claiming token exposed is of another account of mine which is personal and the email received is on work account which are not connected by any medium . But what surprising for me was the email was not received on the personal email id which contains the repo with exposed token and has received on work account. I have only one forked repo of personal account to work account . Also 2nd surprise was that i had never added git guardian to my account, as it is a 3rd party add-on it need to be added. 
Please help me understanding this .Wil be kind full for each reply.
Thanks
![image](https://github.com/user-attachments/assets/71591881-4de1-43d8-8947-107abb66b8b6)

**Reply by Chukwusombiri** [reply]
i dont think thats from gitguarding because it originated from a false domain getgitguardian.com where gitguard's domain is gitguardian.com

**Reply by HenraL** [reply]
I know the thread a bit old, but having received the e-mail myself and looking at the links, it appears that getgitguardian is there e-mail domain name because the links that the user is encouraged to interract with are [dashboard.gitguardian.com](ttps://dashboard.gitguardian.com/) (appart for the unsubscribe that is [email.getgitguardian.com](https://email.getgitguardian.com)).

That being said, you should **ALWAYS** check the links and look up the company when you receive e-mails from known (and especially unknown) senders.

**Comment by estahn** [comment]
@vinyashegde You can clean your repository with something like https://rtyley.github.io/bfg-repo-cleaner/

**Comment by Mohataseem89** [comment]
It looks like the key point here is verifying the sender domain and checking your repositories directly, GitGuardian is a legitimate service, but phishing attempts often mimic real security alerts. If the email came from getgitguardian.com instead of gitguardian.com it's definitely worth treating as suspicious.

the safest approach is to ignore the email link, manually inspect your repos for exposed secrets, and rotate any keys if needed. also good to remember that third-party scanners can flag repos even if you haven’t explicitly installed them, depending on public visibility. always better to verify than trust the email blindly
